src/lib.rs

```diff
@@ -364,6 +364,10 @@
 
 #![cfg_attr(not(feature = "std"), no_std)]
 
+// NOTE: This is a temporary workaround for Rust compiler issue:
+// - https://github.com/rust-lang/rust/issues/129031
+#![allow(unreachable_patterns)]
```

Comment:
I am thinking it may be better to keep this per match statement where this is needed, along with these comments (should probably reference both rust-lang issues), in a similar fashion to this update on `async-lock`: https://github.com/smol-rs/async-lock/pull/92

Reply:
I don't think that's the right approach -- this crate does not officially support nightly Rust, and I don't want to pick up maintainance burden to be compatible with nightlies. 

So, there should not be source-level work-arounds for nightly bugs.

Reply:
How should we move forward then?

Reply:
I think using `RUSTFLAGS=\"-A unreachable_patterns\"` should be able to help, trying in my own fork for now

Reply:
Using `sh.push_env("RUSTFLAGS", "-A unreachable_patterns");` in the `TEST_BETA` does seem to fix the CI building without affecting any of the source code, and I did make a similar comment in issue #261.